File: Roman_number_to_integer/code.cpp
class Solution {
  public:
    int romanToDecimal(string &str) {
        // code here
        int num = 0;
        for(int i=str.size()-1; i>=0; i--){
            if(str[i] == 'I'){
                num += 1;
            }
            else if(str[i] == 'V'){
                num += 5;
                if(str[i-1] == 'I'){
                    num -= 1;
                    i--;
                }
            }
            else if(str[i] == 'X'){
                num+=10;
                if(str[i-1] == 'I'){
                    num -= 1;
                    i--;
                }
            }
            else if(str[i] == 'L'){
                num+=50;
                if(str[i-1] == 'X'){
                    num -= 10;
                    i--;
                }
            }
            else if(str[i] == 'C'){
                num += 100;
                if(str[i-1] == 'X'){
                    num -= 10;
                    i--;
                }
            }
            else if(str[i] == 'D'){
                num += 500;
                if(str[i-1] == 'C'){
                    num -= 100;
                    i--;
                }
            }
            else if(str[i] == 'M'){
                num += 1000;
                if(str[i-1] == 'C'){
                    num -= 100;
                    i--;
                }
            }
        }
        return num;
    }
};

File: Roman_number_to_integer/code.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """

        roman_to_int = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        }

        num = 0
        for i in range(len(s) - 1):
            # Check if the current value is less than the next
            if roman_to_int[s[i]] < roman_to_int[s[i + 1]]:
                # If IV, then it's 5 - 1, which is 4
                num -= roman_to_int[s[i]]
            else:
                num += roman_to_int[s[i]]

        return num + roman_to_int[s[-1]]